core/src/main/java/org/apache/gravitino/auxiliary/AuxiliaryServiceManager.java

```diff
@@ -219,4 +222,43 @@ public void serviceStop() throws Exception {
       throw firstException;
     }
   }
+
+  // Extract aux service configs, transform gravitino.$serviceName.key to $serviceName.key.
+  // And will transform gravitino.auxService.$serviceName.key to $serviceName.key to keep
+  // compatibility.
+  @VisibleForTesting
+  static Map<String, String> extractAuxiliaryServiceConfigs(Config config) {
+    String auxServiceNames =
+        config
+            .getConfigsWithPrefix(GRAVITINO_AUX_SERVICE_PREFIX)
+            .getOrDefault(AUX_SERVICE_NAMES, "");
+    Map<String, String> serviceConfigs = new HashMap<>();
+    serviceConfigs.put(AUX_SERVICE_NAMES, auxServiceNames);
+    config
+        .getAllConfig()
+        .forEach(
+            (k, v) -> {
+              if (k.startsWith(GRAVITINO_AUX_SERVICE_PREFIX)) {
+                serviceConfigs.put(k.substring(GRAVITINO_AUX_SERVICE_PREFIX.length()), v);
+              }
+            });
+    splitter
+        .omitEmptyStrings()
+        .trimResults()
+        .splitToStream(auxServiceNames)
+        .forEach(
+            name ->
+                config.getAllConfig().forEach((k, v) -> extractConfig(serviceConfigs, name, k, v)));
+    return serviceConfigs;
+  }
```

Comment:
I think you should:

1. Update the server config doc to describe the new config names and declare that old keys are still workable.
2. Here in the code, when old configs are configured, output a warn log to users.

Reply:
1. will use separate PR to add document
2. done